Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderAssetBuilder.cpp

```diff
@@ -96,22 +182,17 @@ namespace AZ
                 return;
             }
 
-
-            GlobalBuildOptions buildOptions = ReadBuildOptions(ShaderAssetBuilderName);
-
-            // [GFX TODO] [ATOM-14966] In principle, based on macro definitions, included files can change per supervariant.
-            //                         So, the list of source asset dependencies must be collected by running MCPP on each supervariant.
-            //                         For now, we will run MCPP only once because CreateJobs() should be as light as possible.
-            // 
-            // Regardless of the PlatformInfo and enabled ShaderPlatformInterfaces, the azsl file will be preprocessed
-            // with the sole purpose of extracting all included files. For each included file a SourceDependency will be declared.
-            PreprocessorData output;
-            buildOptions.m_compilerArguments.Merge(shaderSourceData.m_compiler);
-            PreprocessFile(azslFullPath, output, buildOptions.m_preprocessorSettings, true, true);
-            for (auto includePath : output.includedPaths)
+            auto outcome = GetListOfIncludedFiles(azslFullPath);
+            if (!outcome.IsSuccess())
+            {
+                AZ_Error(ShaderAssetBuilderName, false, outcome.GetError().c_str());
+                response.m_result = AssetBuilderSDK::CreateJobsResultCode::Failed;
+                return;
+            }
+            const auto listOfRelativePaths = outcome.TakeValue();
+            for (auto includePath : listOfRelativePaths)
```

Comment:
The above code appears to be using the GetListOfIncludedFiles function (which is meant for regexing includes).  Is this correct?  I'd assume instead you'd want to instead read the json file you're currently parsing for its own includes instead of using #include regex function.  

Then you'd only emit the direct dependencies - ie, azsl files themselves, not the contents of those files?

Reply:
You are right in this case the only dependency is the azsl file that appears in the json file. No need to parse the azsl file. Will change that.